File: frontend/src/api/critical-disposition.ts

import { request } from './client';
import type { CriticalDisposition } from '../types/domain';

export interface ConfirmDispositionInput {
  expectedVersion: number;
  receiveTarget: string;
  dispositionAction: string;
  reason: string;
}

export async function listDispositions(page = 1, pageSize = 50, status = '') {
  const suffix = status ? `&status=${encodeURIComponent(status)}` : '';
  return request<CriticalDisposition[]>(`/dispositions?page=${page}&pageSize=${pageSize}${suffix}`);
}

export async function confirmDisposition(id: number, input: ConfirmDispositionInput) {
  return request<CriticalDisposition>(`/dispositions/${id}/confirm`, {
    method: 'POST', body: JSON.stringify(input),
  });
}

File: backend/internal/dto/critical_disposition.go
package dto

// ConfirmCriticalDisposition 是危急处置事项确认的写入契约。确认人身份来自会话，
// 不由客户端指定。
type ConfirmCriticalDisposition struct {
	ExpectedVersion   uint   `json:"expectedVersion" binding:"required"`
	ReceiveTarget     string `json:"receiveTarget" binding:"required,min=2,max=200"`
	DispositionAction string `json:"dispositionAction" binding:"required,min=4,max=1000"`
	Reason            string `json:"reason" binding:"required,min=3,max=500"`
}
